legend-engine-xt-relationalStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/result/RealizedRelationalResult.java

```diff
@@ -56,9 +57,9 @@ public RealizedRelationalResult(RelationalResult relationalResult) throws SQLExc
         {
             while (resultSet.next())
             {
-                if (rowCount > 1000)
+                if (rowCount > getRowLimit())
```

Comment:
calling this on each iteration will be expensive given the look into the properties map + parsing from string.

Please cache this into a field/variable.